### ISSUE TITLE
Update on use-cases

### ISSUE THREAD

**Issue by jaipradeesh** [upspeak/upspeak#52]
Based on the call me and Kasutav had, we've updated the product to fit in certain specific use-cases which helps us move in right direction and make better product decisions. Following is a bare minimum gist.

- Teams could have multiple threads.
- When a call is started, a thread is created.
- Permissions are defined at team level.
- Member of a team can operate on all the workspaces.
- There could be private workspaces? (Question: Was not part of the discussion.)
- Will think about admin later.
- Private (to the team), public to the teams, public to the world. (Need clarity.)
- Couple major integrations should be part of the initial release (Github, ..)
- A upspeak user can invite any user (via email if user doesn't exist, via username/email if the user exists) to be a part of workspace.
- A call could be joined by users who are not part of upspeak. But any other actions could be done only after signing up as a upspeak user. Similar to Github's model.

Conditions:
- Root view (post login) should be a dashboard. Items and their placements TBD.
- Any action which is to be done (or) upcoming has to be achieved within one click.
- Any recent history access (or) actions on workspaces should be accessible in two clicks.

> In a bit of hurry. Might have missed couple things. @kaustavdm Fill 'em up?

**Comment by kaustavdm** [comment]
Superceded by #67, with inputs from the [comment above](https://github.com/upspeak/upspeak/issues/52#issuecomment-362494129).
